ui/src/app/subnets/views/SubnetDetails/SubnetSummary/SubnetSummaryForm/SubnetSummaryFormFields/SubnetSummaryFormFields.tsx

```diff
@@ -73,23 +46,12 @@ const SubnetSummaryFormFields = ({
           name="allow_dns"
           type="checkbox"
         />
-        {values.fabric && (
-          <FabricSelect
-            name="Fabric"
-            defaultOption={{
-              label: getFabricDisplay(fabric) || "",
-              value: values?.fabric.toString() || "",
-            }}
-          />
-        )}
+        <FabricSelect name="fabric" />
         <VLANSelect
+          fabric={Number(values.fabric)}
           name="vlan"
+          setDefaultValueFromFabric
           showSpinnerOnLoad
-          fabric={vlan?.fabric}
-          defaultOption={{
-            label: getVLANDisplay(vlan) || "",
-            value: values?.vlan.toString(),
-          }}
```

Comment:
This was a bit trickier to figure out. Basically, there's some internal logic in `VLANSelect` to automatically render a list of VLANs based on a given fabric. The way it does that is via the `fabric` prop, which expects a fabric ID (type === `number`). However, in the `FabricSelect` component it sets that ID to a string, so there is actually some inconsistency between what we've typed the form values to be and what they actually are. I'm not certain why the IDs are coerced to be strings (@petermakowski might be able to answer that), but for the form values to be the right type, `fabric` and `vlan` should both be `string`. And since that's the case, we have to convert it *back* to `number` when we pass it to the `VLANSelect` component, because otherwise you'll never find a fabric with e.g `id === "1"`.

I'm happy to jump on a call when you get in if I'm not being very clear. :+1: